### ISSUE TITLE
Accessibilty Issue over Theme and Language Dropdowns

### ISSUE THREAD

**Issue by at-the-vr** [withastro/starlight#1251]
### What version of `starlight` are you using?

0.15.0

### What version of `astro` are you using?

4.0.1

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### What browser are you using?

Chrome

### Describe the Bug

According to `Dev Toolbar Audit` the labels used for `ThemeSelect`& `LanguageSelect` are missing its controlled text
Complete Error according to Dev Toolbar ⬇
```js
"label" tag should have an associated control and a text content.`
The label tag must be associated with a control using either for or having a nested input. Additionally, the label tag must have text content.
```
Link to component found from Audit: [Select Component](https://github.com/withastro/starlight/blob/main/packages/starlight/components/Select.astro)

Screenshots ⬇️

![image](https://github.com/withastro/starlight/assets/88548999/bc9a2d0b-4cea-4c0d-8525-dffd8fd99239)
![image](https://github.com/withastro/starlight/assets/88548999/0e7d3345-b00d-43ca-87d9-16429438a6fa)

> Note: The Stackblitz instance for [Starlight](https://astro.new/starlight-basics?on=stackblitz) does not highlights this issue but any local starlight repo will show this issue

### Link to Minimal Reproducible Example

Not applicable

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by at-the-vr** [comment]
I think it looks weird how the Audit message is being chopped up under normal viewport and looks fine by  reducing the tab width via Chrome DevTools, I do wanna know how to type this Issue out to `withastro/astro` without sounding ambiguous

**Comment by HiDeoo** [comment]
Thanks for the report.

For more information, the rule details are located [here](https://github.com/withastro/astro/blob/main/packages/astro/src/runtime/client/dev-overlay/plugins/audit/a11y.ts#L282-L292).

```js
{
  code: 'a11y-label-has-associated-control',
  title: '`label` tag should have an associated control and a text content.',
  message:
    'The `label` tag must be associated with a control using either `for` or having a nested input. Additionally, the `label` tag must have text content.',
  selector: 'label:not([for])',
  match(element) {
    const inputChild = element.querySelector('input');
    if (!inputChild?.textContent) return true;
  },
},
```

It looks like that after matching the `label:not([for])` CSS selector, it's looking inside for only `input` elements, but I think it should also look for [`meter`, `output`, `progress`, `select`, and `textarea` elements](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/label-has-associated-control.js#L65-L72) as well.

This may need a second pair of eyes but my initial thinking is that this is a false positive that should be fixed upstream.

> I think it looks weird how the Audit message is being chopped up under normal viewport and looks fine by reducing the tab width via Chrome DevTools, I do wanna know how to type this Issue out to `withastro/astro` without sounding ambiguous

This one has already been reported in https://github.com/withastro/astro/issues/9346

**Comment by delucis** [comment]
I agree with @HiDeoo that this is a false positive and should be fixed in the Astro repo. Just as extra confirmation, neither Firefox’s built-in a11y tests nor axe Dev Tools report this issue, and looking at the accessibility tree for the language switcher (for example) shows labels working as expected:

<img width="182" alt="Screenshot of the accessibility tree inspector showing a name of “Select language” and value of “English”" src="https://github.com/withastro/starlight/assets/357379/d5dd58dc-5bd9-4a1c-ad1e-0ce137366fd4">

I will transfer this over to the `astro` repo and add some extra context.
